rpc/serve.go

```diff
@@ -0,0 +1,72 @@
+package rpc
+
+import (
+	"context"
+	"errors"
+	"net"
+
+	"capnproto.org/go/capnp/v3"
+)
+
+// Serve serves a Cap'n Proto RPC to incoming connections.
+//
+// Serve will take ownership of bootstrapClient and release it after the listener closes.
+//
+// Serve exits with the listener error if the listener is closed by the owner.
+func Serve(lis net.Listener, bootstrapClient capnp.Client) error {
+	if !bootstrapClient.IsValid() {
+		err := errors.New("BootstrapClient is not valid")
+		return err
+	}
+	// Accept incoming connections
+	for {
+		rwc, err := lis.Accept()
+		if err != nil {
+			// Since we took ownership of the bootstrap client, release it after we're done.
```

Comment:
I would instead just put `defer bootstrapClient.Release()` at the top of the function.